### PR TITLE
feat: add clone action

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* adds clone action to TableActions
* handles if we're trying to clone in EditView (publish is temp disabled)
* adds E2E for cloning since it's a complicated feature across multiple views and parameters

### How to test it?

* As a user I want to clone a document from the list-view

### Related issue(s)/PR(s)

* resolves CONTENT-1892

### CODE REVIEW

File: e2e/tests/content-manager/cloning.spec.ts
Comment (joshuaellis) [inline e2e/tests/content-manager/cloning.spec.ts:1]: @MarionLemaire

File: packages/core/admin/admin/src/content-manager/hooks/useDocumentActions.ts [hunk 1]
```diff
@@ -395,6 +405,39 @@ const useDocumentActions: UseDocumentActions = () => {
...
+
+        if ('error' in res) {
+          toggleNotification({ type: 'warning', message: formatAPIError(res.error) });
+
+          return { error: res.error };
+        }
+
+        toggleNotification({
+          type: 'success',
+          message: { id: getTranslation('success.record.save') },
```
Comment (jhoward1994) [inline packages/core/admin/admin/src/content-manager/hooks/useDocumentActions.ts:425]: Would a more descriptive success message be better for cloning?

File: packages/core/admin/admin/src/content-manager/pages/EditView/components/DocumentActions.tsx [hunk 2]
```diff
@@ -552,6 +563,23 @@ const UpdateAction: DocumentActionComponent = ({ activeTab, id, model, collectio
...
+          );
+
+          if ('data' in res) {
+            /**
+             * TODO: refactor the router so we can just do `../${res.data.id}` instead of this.
+             */
+            navigate({
+              pathname: `../${collectionType}/${model}/${res.data.id}`,
+            });
+          }
```
Comment (jhoward1994) [inline packages/core/admin/admin/src/content-manager/pages/EditView/components/DocumentActions.tsx:582]: I'm not sure but should the isCloning condition come first? Is there any case
when we're on the clone route that we would want to call the normal update?

**Comment by jhoward1994** [comment]
Screenshot 2024-02-19 at 12 29 04 (2)

Is it necessary to have the warning and modal? They seem to explain the same thing?

**Comment by jhoward1994** [comment]
Uploading Screen Recording 2024-02-19 at 12.31.28.mov…

I think this a BE thing, validations don't seem to be running in the same way on the clone route

e.g. unique validations will apply correctly when creating/updating an entry but not when duplicating an entry

File: e2e/tests/content-manager/cloning.spec.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
+import { test, expect } from '@playwright/test';
+import { resetDatabaseAndImportDataFromPath } from '../../scripts/dts-import';
+import { login } from '../../utils/login';
+import { pageSizes } from '../../../packages/core/upload/admin/src/constants';
```
Comment (jhoward1994) [inline e2e/tests/content-manager/cloning.spec.ts:4]: Remove this line?

File: packages/core/admin/admin/src/content-manager/hooks/useDocumentActions.ts (hunk 1 above)
Comment (joshuaellis) [reply packages/core/admin/admin/src/content-manager/hooks/useDocumentActions.ts:425]: Good thinking!

File: e2e/tests/content-manager/cloning.spec.ts (hunk 3 above)
Comment (joshuaellis) [reply e2e/tests/content-manager/cloning.spec.ts:4]: oh yeah, i wonder how that got there!?

File: packages/core/admin/admin/src/content-manager/pages/EditView/components/DocumentActions.tsx (hunk 2 above)
Comment (joshuaellis) [reply packages/core/admin/admin/src/content-manager/pages/EditView/components/DocumentActions.tsx:582]: I'm not sure it should make a huge difference, but cognitively it probably makes more sense the way round you're proposing. Lets do that.

**Comment by joshuaellis** [comment]
> Is it necessary to have the warning and modal? They seem to explain the same thing?

I think it's better to keep it, the notification is owned by the action runner. We implement the modal as part of the CM. I don't think it's the end of the world to have both, I think notifications are announced too 🤔

**Comment by joshuaellis** [comment]
> I think we should add an E2E case for this. Shall I work on modifying the clone route in a new PR?

Yeah separate PR would be best imo.

File: packages/core/admin/admin/src/translations/en.json [hunk 4]
```diff
@@ -680,6 +680,8 @@
   "components.popUpWarning.message": "Are you sure you want to delete this?",
   "components.popUpWarning.title": "Please confirm",
   "content-manager.App.schemas.data-loaded": "The schemas have been successfully loaded",
+  "content-manager.actions.clone.error": "An error occurred while trying to clone the document.",
```
Comment (jhoward1994) [inline packages/core/admin/admin/src/translations/en.json:683]: Is it weird that in one place we say 'duplicate' and somewhere else say 'clone'?

**Review by jhoward1994 — APPROVED** [review]

**Pull request merged**